modules/tracking/samples/lasot_benchmark.py

```diff
@@ -0,0 +1,193 @@
+            for number_of_the_frame, image in enumerate(video_sequence):
+
+                for i in range(len(gt_bb)):
+                    gt_bb[i] = float(gt_bb[i])
+                gt_bb = tuple(gt_bb)
+                frame = cv.imread(image)
+
+                # Condition of tracker`s re-initialization
+                if ((number_of_the_frame + 1) % frames_before_reinit == 0) and (
+                        number_of_the_frame != 0):
+
+                    tracker, frames_before_reinit = init_tracker(tracker_name)
+                    init_once = False
+                    init_bb = gt_bb
+
+                if not init_once:
+                    init_state = tracker.init(frame, init_bb)
+                    init_once = True
+                init_state, new_bb = tracker.update(frame)
+
+                # Check for presence of object on the frame
+                # If no object on frame, we reduce number of
+                # accounted for evaluation frames
+                if gt_bb != (0, 0, 0, 0):
+                    # Calculation of coordinates of corners and centers 
+                    # from [x, y, w, h] bounding boxes
+
+                    new_coords = calc_coords(new_bb)
+                    gt_coords = calc_coords(gt_bb)
+                    new_xmin, new_xmax, new_ymin, new_ymax, new_cx, new_cy = list(
+                        new_coords.values())
+                    gt_xmin, gt_xmax, gt_ymin, gt_ymax, gt_cx, gt_cy = list(gt_coords.values())
+
+                    if args.visualization:
+                        cv.rectangle(frame, (int(new_xmin), int(new_ymin)), (
+                            int(new_xmax), int(new_ymax)), (200, 0, 0))
+                        cv.imshow("Tracking", frame)
+                        cv.waitKey(1)
+                    # Metrics: Intersection over Union, Precision, 
+                    # Normalized Precision
+
+                    # Width and height of overlap
+                    dx = max(0, min(new_xmax, gt_xmax) - max(
+                        new_xmin, gt_xmin))
+                    dy = max(0, min(new_ymax, gt_ymax) - max(
+                        new_ymin, gt_ymin))
+
+                    # Intersection over Union
+                    area_of_overlap = dx * dy
+                    area_of_union = (new_xmax - new_xmin) * (
+                        new_ymax - new_ymin) + (gt_xmax - gt_xmin) * (
+                            gt_ymax - gt_ymin) - area_of_overlap
+                    # Zero division check
+                    if area_of_union != 0:
+                        iou = area_of_overlap / area_of_union
+                        sum_iou += iou
+
+                    # Precision
+                    precision = np.sqrt((new_cx - gt_cx) ** 2 + (
+                        new_cy - gt_cy) ** 2)
+                    if precision > 20.0:
+                        pr_value = 0.0
+                    else:
+                        pr_value = 1.0
+                    sum_pr += pr_value
```

Comment:
Maybe create separate functions for calculating IoU and precision metrics?

Reply:
Here is no code duplication, so I do not see a reason to do that.

Reply:
For me, this makes main function more readable and we don't need comments like:
```
# Intersection over Union
# Precision
# Normalized precision
```

`iou = get_iou(new_coords, gt_coords)`

Reply:
IMHO clear comments and variable names are better for readability of code. These calculations are not that hard. Hiding 7-8 lines of nonrepeating code in the separate method can cause more difficulties for the user. I do not understand why we should hide calculations like this.

Reply:
We need to separate the logic from implementation. That's why using methods we can show what's programmed.

Reply:
OK, but you want to leave normalized precision as is, am I correct?

Reply:
I add a method for every metric.